### PR TITLE
JAMES-2947 WebAdmin prevents encoded '/' in a domain name at creation

### PR DESCRIPTION
**Pull request by blackheaven**

### CODE REVIEW

**Comment by blackheaven** [comment]
> that nice to fix the case for '/' but did you check for other forbidden characters in the RFC? Did you also check that newer RFC are not allowing it?

In fact, in the root RFC, only alphanumeric values, dots and "-" are allowed

**Comment by Arsnael** [comment]
CI restart...
test this please

**Comment by Arsnael** [comment]
No logs...

Test this please

**Comment by Arsnael** [comment]
CI restart...

test this please

**Comment by blackheaven** [comment]
No logs

test this please

File: core/src/main/java/org/apache/james/core/Domain.java [hunk 1]
```diff
@@ -38,17 +40,19 @@ private static String removeBrackets(String domainName) {
 
     public static Domain of(String domain) {
         Preconditions.checkNotNull(domain, "Domain can not be null");
-        Preconditions.checkArgument(!domain.isEmpty() && !domain.contains("@"),
-            "Domain can not be empty nor contain `@`");
-        return new Domain(domain);
+        String normalizedDomainName = removeBrackets(domain.toLowerCase(Locale.US));
+        Preconditions.checkArgument(InternetDomainName.isValid(normalizedDomainName) || InetAddresses.isInetAddress(normalizedDomainName),
```
Comment (rouazana) [inline core/src/main/java/org/apache/james/core/Domain.java]: what about validating it before lowercasing it?

**Comment by chibenwa** [comment]
test this please

**Comment by Arsnael** [comment]
```
testCompile) on project james-server-webadmin-data: Compilation failure: Compilation failure: 
[e42d349473ceeb9b52d6942da41d50cc5fcfa382] [ERROR] /james-project/server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainsRoutesTest.java:[158,13] cannot find symbol
[e42d349473ceeb9b52d6942da41d50cc5fcfa382] [ERROR]   symbol:   class Map
[e42d349473ceeb9b52d6942da41d50cc5fcfa382] [ERROR]   location: class org.apache.james.webadmin.routes.DomainsRoutesTest.NormalBehaviour
[e42d349473ceeb9b52d6942da41d50cc5fcfa382] [ERROR] /james-project/server/protocols/webadmin/webadmin-data/src/test/java/org/apache/james/webadmin/routes/DomainsRoutesTest.java:[176,13] cannot find symbol
[e42d349473ceeb9b52d6942da41d50cc5fcfa382] [ERROR]   symbol:   class Map
[e42d349473ceeb9b52d6942da41d50cc5fcfa382] [ERROR]   location: class org.apache.james.webadmin.routes.DomainsRoutesTest.NormalBehaviour
[e42d349473ceeb9b52d6942da41d50cc5fcfa382] [ERROR] -> [Help 1]
```

**Comment by Arsnael** [comment]
```
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] [ERROR]   DLPConfigurationRoutesTest$DefineClear.deleteShouldReturnBadRequestWhenDomainIsNotValid:646 1 expectation failed.
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] JSON path details doesn't match.
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] Expected: is "Domain can not be empty nor contain `@`"
[2e625f2cec63106ee6be5bfef1d602ddd305afd4]   Actual: Domain can not be empty nor contain `@` nor `/`
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] 
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] [ERROR]   DLPConfigurationRoutesTest$DefineList.getShouldReturnBadRequestWhenDomainIsNotValid:859 1 expectation failed.
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] JSON path details doesn't match.
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] Expected: is "Domain can not be empty nor contain `@`"
[2e625f2cec63106ee6be5bfef1d602ddd305afd4]   Actual: Domain can not be empty nor contain `@` nor `/`
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] 
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] [ERROR]   DLPConfigurationRoutesTest$DefineStore.putShouldReturnBadRequestWhenDomainIsNotValid:329 1 expectation failed.
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] JSON path details doesn't match.
[2e625f2cec63106ee6be5bfef1d602ddd305afd4] Expected: is "Domain can not be empty nor contain `@`"
[2e625f2cec63106ee6be5bfef1d602ddd305afd4]   Actual: Domain can not be empty nor contain `@` nor `/`
```

**Comment by Arsnael** [comment]
```
[ERROR]   WithEncryptedAwsS3Test.mailsShouldBeWellReceived » ConditionTimeout Condition ...
```
test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**